ghproxy/ghmetrics/ghpath.go

```diff
@@ -0,0 +1,424 @@
+						children: []node{{
+							Node: variable("username"),
+						}},
+					},
+					{Node: literal("repos")},
+					{Node: literal("hovercard")},
+					{Node: literal("following")},
+				},
+			}},
+		}
+	case "orgs":
+		// /orgs
+		tree = node{
+			Node: literal("orgs"),
+			children: []node{{
+				Node: variable("orgname"),
+				children: []node{
+					{
+						Node: literal("credential-authorizations"),
+						children: []node{{
+							Node: variable("credentialId"),
+						}},
+					},
+					{Node: literal("repos")},
+					{Node: literal("issues")},
+				},
+			}},
+		}
+	case "issues":
+		// /issues
+		tree = node{
+			Node: literal("issues"),
+			children: []node{{
+				Node: variable("issueId"),
+			}},
+		}
+	case "search":
+		// do we care to handle search sub-paths differently?
+		// e.g.: /search/repositories, /search/commits, /search/code, /search/issues, /search/users, /search/topics, /search/labels
+		tree = node{
+			Node: literal("search"),
+			children: []node{
+				{Node: literal("repositories")},
+				{Node: literal("commits")},
+				{Node: literal("code")},
+				{Node: literal("issues")},
+				{Node: literal("users")},
+				{Node: literal("topics")},
+				{Node: literal("labels")},
+			},
+		}
+	case "gists":
+		// do we care to handle gist sub-paths differently?
+		// e.g. /gists/public, /gists/starred
+		tree = node{
+			Node: literal("gists"),
+			children: []node{
+				{Node: literal("public")},
+				{Node: literal("starred")},
+			},
+		}
+	case "notifications":
+		// /notifications
+		tree = node{
+			Node: literal("notifications"),
+			children: []node{
+				{
+					Node: literal("threads"),
+					children: []node{{
+						Node: variable("threadId"),
+						children: []node{{
+							Node: literal("subscription"),
+						}},
+					}},
+				},
+			},
+		}
+	case "repositories", "emojis", "events", "feeds", "hub", "rate_limit", "teams", "licenses":
+		tree = node{
+			Node: literal(fragment),
+		}
+	default:
+		logrus.WithField("path", path).Warning("Path not handled")
+		return path
+	}
+
+	splitPath := strings.FieldsFunc(path, splitFunc)
+	resolvedPath, err := handlePath(tree, splitPath)
+	if err != nil {
+		logrus.WithField("path", path).Warning("Path not handled")
+		return path
+	}
+	return fmt.Sprintf("/%s", resolvedPath)
+}
+
+func splitFunc(c rune) bool {
+	return c == '/'
+}
+
+func handlePath(tree node, path []string) (string, error) {
+	representation, matches := resolve(tree, path)
+	if !matches {
+		errMsg := fmt.Sprintf("no match for path " + strings.Join(path, "/"))
+		return "", fmt.Errorf(errMsg)
+	}
+	return representation, nil
+}
+
+// getFirstFragment returns the first fragment of a path, of a
+// `*url.URL.Path`. e.g. `/repos/kubernetes/test-infra/` will return `repos`
+func getFirstFragment(path string) string {
+	result := strings.Split(path, "/")
+	for _, str := range result {
+		if str != "" {
+			return str
+		}
+	}
+	return ""
+}
+
+type node struct {
+	Node
```

Comment:
I'd suggest renaming `Node` to something that indicates that it is data of some sort; such as `data` or `urlFragment`

While a `node` is a node in a tree that has children, a `Node` can't have children, or be a child, or be a node in the graph at all; only `node`s can